function/src/main/proto/message.c

```diff
@@ -9,20 +9,39 @@ typedef struct _wrapper
 
 Wrapper filter(Wrapper input)
 {
-    SpringMessage *msg = spring_message__unpack(NULL, input.len, input.data);
-    SpringMessage *result = malloc(sizeof(SpringMessage));
-    spring_message__init(result);
-    result->payload = msg->payload;
-    result->headers = msg->headers;
-    result->n_headers = msg->n_headers;
-    int len = spring_message__get_packed_size(result);
+    Io__Cloudevents__V1__CloudEvent *msg = io__cloudevents__v1__cloud_event__unpack(NULL, input.len, input.data);
+    Io__Cloudevents__V1__CloudEvent *result = malloc(sizeof(Io__Cloudevents__V1__CloudEvent));
+    io__cloudevents__v1__cloud_event__init(result);
+
+    result->id = "1-5150"; //msg->id; TODO: see below - decorate by adding an attribute, not modify the id
+    result->source = msg->source;
+    result->spec_version = msg->spec_version;
+    result->type = msg->type;
+	result->data_case = msg->data_case;
+	result->binary_data = msg->binary_data;
+	result->text_data = msg->text_data;
+	result->proto_data = msg->proto_data;
+	/*
+	TODO: Figure out how to add another attribute 'decoratedby: "wasm"' (need to refamiliarize w/ '**' pointer-to-pointer usage.
+	Io__Cloudevents__V1__CloudEvent__CloudEventAttributeValue newAttrValue = IO__CLOUDEVENTS__V1__CLOUD_EVENT__CLOUD_EVENT_ATTRIBUTE_VALUE__INIT;
+	newAttrValue->attr_case = IO__CLOUDEVENTS__V1__CLOUD_EVENT__CLOUD_EVENT_ATTRIBUTE_VALUE__ATTR_CE_STRING;
+	newAttrValue->ce_string = "wasm"
+	Io__Cloudevents__V1__CloudEvent__AttributesEntry newAttrEntry = IO__CLOUDEVENTS__V1__CLOUD_EVENT__ATTRIBUTES_ENTRY__INIT;
+	newAttrEntry->key = "decoratedby";
+	newAttrEntry->value = newAttrValue;
+
+	result->n_attributes = (msg->n_attributes)+1;
+	result->attributes = msg->attributes;
+	*/
+	result->n_attributes = msg->n_attributes;
+	result->attributes = msg->attributes;
+
+    int len = io__cloudevents__v1__cloud_event__get_packed_size(result);
     uint8_t *buffer = malloc(len);
-    spring_message__pack(result, buffer);
-    spring_message__free_unpacked(msg, NULL);
+    io__cloudevents__v1__cloud_event__pack(result, buffer);
+    io__cloudevents__v1__cloud_event__free_unpacked(msg, NULL);
     Wrapper output = {
         buffer,
         len};
     return output;
 }
-
-// $ emcc -I ../include -s ERROR_ON_UNDEFINED_SYMBOLS=0 -Os -s STANDALONE_WASM -s EXPORTED_FUNCTIONS="['_filter','_malloc','_free']" -Wl,--no-entry message.c message.pb-c.c ../lib/libprotobuf-c.a ../lib/libprotobuf.a -o message.wasm
```

Comment:
Since we have this in the README and in the Makefile I figured it was safe to pull out of here as it was getting out of date.